pkg/lib/routines/routines.go

```diff
@@ -0,0 +1,35 @@
+/*
+Copyright 2020 Cortex Labs, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package routines
+
+import (
+	"fmt"
+
+	"github.com/cortexlabs/cortex/pkg/lib/exit"
+)
+
+func RunWithPanicHandler(f func()) {
+	go func() {
+		defer func() {
+			if r := recover(); r != nil {
+				err := ErrorUnexpectedError(fmt.Sprintf("%v", r))
+				exit.Error(err)
```

Comment:
I'm not sure if we always want to exit; I think in the operator it would be good to just send the telemetry, print the stack trace, and not exit; that is what we do with the operator API server [here](https://github.com/cortexlabs/cortex/blob/master/pkg/operator/endpoints/middleware.go#L43) ([implementation](https://github.com/cortexlabs/cortex/blob/master/pkg/operator/endpoints/respond.go#L71)) and in the cron ([here](https://github.com/cortexlabs/cortex/blob/master/pkg/lib/cron/cron.go#L72)). Perhaps in the CLI exiting makes sense. What do you think?

Reply:
As per our last conversation, we are no longer exiting if a panic occurs and instead we recover and report the error.